ts/utils/ipex_optimization.py

```diff
@@ -0,0 +1,144 @@
+                dummy_tensor = dummy_tensor.to(memory_format=torch.channels_last)
+            x.append(dummy_tensor)
+        return x 
+
+    def warmup_forward(self, m, *x, profiling_count=2):
+        for _ in range(profiling_count):
+            m(*x)
+        return 
+    
+    def get_torchscripted_model(self, m, x, freeze=True):
+        if isinstance(m, torch.nn.Module):
+            m.eval()
+        with torch.no_grad():
+            if not isinstance(m, torch.jit.ScriptModule):
+                try:
+                    traced = torch.jit.trace(m, x)
+                except RuntimeError:
+                    try: 
+                        traced = torch.jit.trace(m, x, check_trace=False, strict=False)
+                    except:
+                        logger.error("Tracing the model failed. Pleace make sure the model is traceable")
+                        exit(-1)
+            else:
+                traced = m
+            
+            if not isinstance(traced, torch.jit.RecursiveScriptModule):
+                freezed = torch.jit.freeze(traced)
+            else:
+                freezed = traced 
+            self.warmup_forward(freezed, *x)
+            
+        return freezed 
+        
+    def optimize(self):
+        if self.onednn_graph_fusion_enabled:
+            torch.jit.enable_onednn_fusion(True)
+        
+        # channel last 
+        if self.channel_last == "true":
+            self.model = self.model.to(memory_format=torch.channels_last)
+        
+        # ipex optimization 
+        if self.ipex_enabled:    
+            import intel_extension_for_pytorch as ipex
+
+            if self.dtype == "float32":
+                self.model = ipex.optimize(self.model, dtype=torch.float32)
+            elif self.dtype == "bfloat16":
+                self.model = ipex.optimize(self.model, dtype=torch.bfloat16)
+            else: # int8
+                from intel_extension_for_pytorch.quantization import prepare, convert
+                from torch.ao.quantization import MinMaxObserver, PerChannelMinMaxObserver, QConfig
+                
+                x = self.get_dummy_tensors()
+                
+                qconfig = QConfig(activation=MinMaxObserver.with_args(qscheme=self.TORCH_QSCHEME[self.qscheme], dtype=torch.quint8),weight=PerChannelMinMaxObserver.with_args(dtype=torch.qint8, qscheme=torch.per_channel_symmetric))
+                self.model = prepare(self.model, qconfig, example_inputs=x, inplace=False)
+                # calibration 
+                n_iter = 100
+                for i in range(n_iter):
+                    self.model(*x)
+                # conversion 
+                self.model = convert(self.model)
+    
+        # torchscript
+        if self.torchscript == "true":
+            x = self.get_dummy_tensors()
+            
+            if self.dtype == "float32" or self.dtype == "int8":
+                self.model = self.get_torchscripted_model(self.model, x)

```

Comment:
Thanks @min-jean-cho for the PR. In the [current version](https://github.com/pytorch/serve/blob/master/ts/torch_handler/base_handler.py#L103), seems ipex optimizations are limited to ipex.optimize(). Can you please elaborate a bit more on the use of torchscript here? would torch scripting the ipex_optimized model help with the perf?

Reply:
Also, the same for torch.cpu.amp.autocast()